src/events/reggelt.ts

```diff
@@ -1,195 +1,231 @@
-                        message.reply("Error: " + error);
-                        console.error("Error:", error);
-                    });
-            }
-            await reggeltupdatefs(message, true);
+    if (
+      message.content.toLowerCase().includes(lang.events.reggelt.keyWord) ||
+      message.content.toLowerCase().includes("reggelt")
+    ) {
+      const ref = db.collection("dcusers").doc(message.author.id);
+      const doc = await ref.get();
+
+      const cdref = db
+        .collection("dcusers")
+        .doc(message.author.id)
+        .collection("cooldowns")
+        .doc(message.guild!.id);
+      const cddoc = await cdref.get();
+
+      const guildcd = guildconfig.cd;
+
+      let rawcd = guildcd;
+
+      const cdval = rawcd * 3600;
+
+      const now = Math.floor(Date.now() / 1000);
+      const cd = now + cdval;
+
+      // on cooldown
+      if (await checkcd(message)) {
+        await message.delete();
+        const nextTime = cddoc.data()!.reggeltcount;
+        let cdmsg = lang.events.reggelt.onCooldown.replace(
+          "%!CD%!",
+          hhmmss(nextTime - now, lang)
+        );
+        await message.author.send(cdmsg);
+      } else {
+        if (
+          !process.env.PROD &&
+          rawcd <= 6 /*&& userdoc.data()!.reggeltcountcd < Date.now()*/
+        ) {
+          // update
+          await reggeltupdateall();
+          await reggeltupdatefs(message);
         }
+        // cd update
+        await cdref
+          .set(
+            {
+              reggeltcount: cd,
+            },
+            { merge: true }
+          )
+          .then(async (d) => {
+            // Reaction add here
+            await react(doc.data()?.reggeltemote, message);
+          });
+      }
+    } else {
+      if (!message.deletable) {
+        message.channel.send(lang.events.reggelt.noPerms).catch((err) => {
+          message.guild!.owner!.send(lang.events.reggelt.noSend);
+          console.error(err);
+        });
+      } else {
+        message.delete();
+        let nReggelt: string = lang.events.reggelt.notReggelt;
+        let replyMSG = nReggelt
+          .replace("%!GUILD%!", `${message.guild!.name}`)
+          .replace("**%!KEYWORD%**", `**${lang.events.reggelt.keyWord}**`);
+        message.author.send(replyMSG).catch(function(error: string) {
+          message.reply("Error: " + error);
+          console.error("Error:", error);
+        });
+      }
+
+      await reggeltupdatefs(message, true);
     }
+  },
+};
+
+async function checkcd(message: Message): Promise<boolean> {
+  const db = admin.firestore();
+  const user = message.author;
+  const uref = db.collection("dcusers").doc(user.id);
+  const cdref = uref.collection("cooldowns").doc(message.guild!.id);
+  const cddoc = await cdref.get();
+  const now = Math.floor(Date.now() / 1000);
+
+  if (!cddoc.exists) {
+    return false;
+  }
+
+  if (cddoc.data()?.reggeltcountcd && cddoc.data()?.reggeltcount > now) {
```

Comment:
Ha a `cddoc.data()?.reggeltcountcd` értéke `null`, akkor a felhasználó tudja spamelni a reggeltet a chaten, és kapja a pontokat. Ha új user jön, nem látom hogy ez az érték neki egyetalán beállításra kerül.

Reply:
`await cdref
          .set(
            {
              reggeltcount: cd,
            },
            { merge: true }
          )`

Reply:
Elvileg az a része elintézi

Reply:
Nem `reggeltcount`, hanem `reggeltcountcd`